airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -87,10 +87,14 @@ class GoogleAnalyticsDataApiBaseStream(GoogleAnalyticsDataApiAbstractStream):
 
     _record_date_format = "%Y%m%d"
     primary_key = "uuid"
-    cursor_field = "date"
+    cursor_field = None
 
     metadata = MetadataDescriptor()
 
+    def __init__(self, *args, config: Mapping[str, Any], **kwargs):
+        self.cursor_field = "date" if "date" in config.get("dimensions") else []
+        super().__init__(*args, config=config, **kwargs)
```

Comment:
Can you walk me through the effects of this on the following previous behavior? 

> The reports are custom by setting the dimensions and metrics required. To support Incremental sync, the `date` dimension is
added by default to all reports. There are 8 default reports. To add more reports, you need to specify the `custom reports` field.

Reply:
sometimes, like here https://github.com/airbytehq/airbyte/issues/15754, users don't want to use `date` in dimensions, so we set cursor field to `date` and therefore allow incremental sync, only if it is presented in config.dimensions.

Reply:
Will the UI still let the user configure an "incremental" sync which will then act as a full refresh because there is no cursor field?

Reply:
If the user doesn't specify the `date` field in dimensions, only a full refresh sync will be available.

Reply:
Maybe we add a note to that affect in the docs [here](https://github.com/airbytehq/airbyte/pull/23872/files#diff-544c6314816bc254aafb18fe1110c690105dc407da4e38384798e251553e6c14R83)? 

For all of the preconfigured streams we are still injecting date as a cursor and using it for incremental syncs, is that correct?

Reply:
All preconfigured streams already have `date` field in dimensions: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/defaults/default_reports.json
Sure, I'll add a note.

Reply:
Cool. Thanks for the context!